src/docfx/lib/schema/JsonSchemaValidator.cs

```diff
@@ -457,6 +468,37 @@ private void ValidateDeprecated(JsonSchema schema, string name, JToken token, Li
             }
         }
 
+        private void ValidateDocsetUnique(JsonSchema schema, string name, JObject map)
+        {
+            foreach (var docsetUniqueKey in schema.DocsetUnique)
+            {
+                if (map.TryGetValue(docsetUniqueKey, out var value))
+                {
+                    var key = string.IsNullOrEmpty(name) ? docsetUniqueKey : $"{name}.{docsetUniqueKey}";
```

Comment:
`(name, docsetUniqueKey)` may not be the correct property identifier, consider use `(schema, docsetUniqueKey)` as the key to identify a property. 

But still need `name` in line 497 to customize error message based on attribute name.

Reply:
some properties defined in the DocsetUnique may not have schema.